src/transaction.ts

```diff
@@ -458,3 +492,215 @@ export class TransactionHash extends Hash {
+
+  /**
+   * The value to transfer.
+   */
+  public value: BigNumber.Value;
+
+  /**
+   * The address of the sender.
+   */
+  public sender: string;
+
+  /**
+   * The address of the receiver.
+   */
+  public receiver: string;
+
+  /**
+   * The username of the sender.
+   */
+  public senderUsername: string;
+
+  /** 
+   * The username of the receiver.
+   */
+  public receiverUsername: string;
+
+  /**
+   * The gas price to be used.
+   */
+  public gasPrice: BigNumber.Value;
+
+  /**
+   * The maximum amount of gas to be consumed when processing the transaction.
+   */
+  public gasLimit: BigNumber.Value;
+
+  /**
+   * The payload of the transaction.
+   */
+  public data: Uint8Array;
+
+  /**
+   * The chain ID of the Network (e.g. "1" for Mainnet).
+   */
+  public chainID: string;
+
+  /**
+   * The version, required by the Network in order to correctly interpret the contents of the transaction.
+   */
+  public version: number;
+
+  /**
+   * The options field of the transactions.
+   */
+  public options: number;
+
+  /**
+   * The address of the guardian.
+   */
+  public guardian: string;
+
+  /**
+   * The signature.
+   */
+  public signature: Uint8Array;
+
+  /**
+   * The signature of the guardian.
+   */
+  public guardianSignature: Uint8Array;
+
+  /**
+   * Creates a new Transaction object.
+   */
+    public constructor({
+      sender,
+      receiver,
+      gasLimit,
+      chainID,
+      nonce,
+      value,
+      senderUsername,
+      receiverUsername,
+      gasPrice,
+      data,
+      version,
+      options,
+      guardian,
+    }: {
+      nonce?: BigNumber.Value;
+      value?: BigNumber.Value;
+      sender: string;
+      receiver: string;
+      senderUsername?: string;
+      receiverUsername?: string;
+      gasPrice?: BigNumber.Value;
+      gasLimit: BigNumber.Value;
+      data?: Uint8Array;
+      chainID: string;
+      version?: number;
+      options?: number;
+      guardian?: string;
+    }) {
+      this.nonce = nonce || 0;
+      this.value = value || new BigNumber(0);
+      this.sender = sender;
+      this.receiver = receiver;
+      this.senderUsername = senderUsername || "";
+      this.receiverUsername = receiverUsername || "";
+      this.gasPrice = gasPrice || new BigNumber(TRANSACTION_MIN_GAS_PRICE);
+      this.gasLimit = gasLimit;
+      this.data = data || new Uint8Array();
+      this.chainID = chainID;
+      this.version = version || TRANSACTION_OPTIONS_DEFAULT;
+      this.options = options || TRANSACTION_OPTIONS_DEFAULT;
+      this.guardian = guardian || "";
+  
+      this.signature = new Uint8Array();
+      this.guardianSignature = new Uint8Array();
+    }
+}
+
+/**
+ * An utilitary class meant to work together with the {@link TransactionNext} class.
+ */
+export class TransactionComputer {
+  constructor() {}
+
+  computeTransactionFee(transaction: ITransactionNext, networkConfig: INetworkConfig): BigNumber{
+    let moveBalanceGas = new BigNumber(
```

Comment:
When possible, we should use `const` instead of `let` (optional, can also be done in other PR).

Reply:
Replaced with const where possible.